src/pylhe/__init__.py

```diff
@@ -225,35 +230,31 @@ def visualize(event, outputname):
     g = nx.DiGraph()
     for i, p in enumerate(event.particles):
         g.add_node(i, attr_dict=p.__dict__)
-        name = pypdt.particle(p.id).name
-        greek = [
-            "gamma",
-            "nu",
-            "mu",
-            "tau",
-            "rho",
-            "Xi",
-            "Sigma",
-            "Lambda",
-            "omega",
-            "Omega",
-            "Alpha",
-            "psi",
-            "phi",
-            "pi",
-            "chi",
-        ]
-        for greekname in greek:
-            if greekname in name:
-                name = name.replace(greekname, "\\" + greekname)
-        if "susy-" in name:
-            name = name.replace("susy-", "\\tilde ")
-        g.node[i].update(texlbl="${}$".format(name))
+        try:
+            iid = int(p.id)
+            if is_SUSY(iid):
+                # for SUSY particles, cf. https://github.com/scikit-hep/pylhe/pull/27#issuecomment-633127984
+                _PDGID2PDGNameMap, _PDGName2PDGIDMap = DirectionalMaps(
```

Comment:
This mapping should be done once and for all, so please remove it from the loop ;-). And it is probably good to `del` these at the end of the file ...

Reply:
The mapping gives you LaTeX names straightaway so I would rather define `_PDGID2LaTeXNameMap, _LaTeXName2PDGIDMap`.